run_refinery.py

```diff
@@ -40,24 +42,35 @@ def process_all_organism_genesets(organism_ini_file, download_folder,
     Python dictionary.
     """
 
+    logger.info('Starting to download all files for organism file %s',
+                organism_ini_file)
     download_all_files(organism_ini_file, download_folder,
                        secrets_location=secrets_file)
+    logger.info('Finished downloading all files for organism file %s',
+                organism_ini_file)
+
     all_genesets = []
 
     species_config_file = SafeConfigParser()
     species_config_file.read(organism_ini_file)
 
-    if species_config_file.has_section('GO'):
-        all_go_terms = process_go_terms(organism_ini_file, download_folder)
-        all_genesets.extend(all_go_terms)
-
-    if species_config_file.has_section('KEGG'):
-        all_kegg_sets = process_kegg_sets(organism_ini_file, download_folder)
-        all_genesets.extend(all_kegg_sets)
-
-    if species_config_file.has_section('DO'):
-        all_do_terms = process_do_terms(organism_ini_file)
-        all_genesets.extend(all_do_terms)
+    annotation_type_dict = {
+        'GO': process_go_terms,
+        'KEGG': process_kegg_sets,
+        'DO': process_do_terms
+    }
+
+    for annot_type, func_name in annotation_type_dict.iteritems():
+        if species_config_file.has_section(annot_type):
+            logger.info('Starting to process %s terms for %s',
+                        annot_type, organism_ini_file)
+            if annot_type == 'DO':
+                processed_sets = func_name(organism_ini_file)
+            else:
+                processed_sets = func_name(organism_ini_file, download_folder)
```

Comment:
I didn't realize that the arguments of function calls are different when the term is different. So the advantage of using a loop is not as big as I expected. It is totally up to you whether you want to use the original code or the loop.

Reply:
Sure - I actually do feel like this code is cleaner now, so I will leave this new version. I think it is easier to read than what I had before.